parquet-hadoop/src/main/java/org/apache/parquet/hadoop/InternalParquetRecordReader.java

```diff
@@ -167,13 +169,13 @@ public float getProgress() throws IOException, InterruptedException {
 
   public void initialize(ParquetFileReader reader, ParquetReadOptions options) {
     // copy custom configuration to the Configuration passed to the ReadSupport
-    Configuration conf = new Configuration();
-    if (options instanceof HadoopReadOptions) {
-      conf = ((HadoopReadOptions) options).getConf();
-    }
+    ParquetConfiguration conf = Objects.requireNonNull(options).getConfiguration();
     for (String property : options.getPropertyNames()) {
       conf.set(property, options.getProperty(property));
     }
+    for (Map.Entry<String, String> property : new Configuration()) {
```

Comment:
CMIW, we do not need to address the Hadoop dependency in the parquet-hadoop module, right?

Reply:
The Hadoop specific stuff I agree that it would be a bit silly. However, this class `InternalParquetRecordReader` is part of the read/write API for which we're trying to address it. That the read/write API is part of parquet-hadoop is somewhat unfortunate, but changing it at this point would break stuff.